### PR TITLE
[BugFix] rewrite column name for mv of aggreagted table

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes https://github.com/StarRocks/StarRocksTest/issues/3823

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by satanson — APPROVED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvRewriteTest.java [hunk 1]
```diff
@@ -1965,4 +1965,36 @@ public void testPlanCache() throws Exception {
...
+        starRocksAssert.withMaterializedView("CREATE MATERIALIZED VIEW mv_t1_v0 " +
+                "AS " +
+                "SELECT t1_17.c_1_0, t1_17.c_1_1, SUM(t1_17.c_1_2) " +
+                "FROM t1_agg AS t1_17 " +
+                "GROUP BY t1_17.c_1_0, t1_17.c_1_1 ORDER BY t1_17.c_1_0 DESC, t1_17.c_1_1 ASC");
+
+        {
+            String query = "select * from t1_agg";
+            String plan = UtFrameUtils.getVerboseFragmentPlan(connectContext, query);
+            PlanTestBase.assertContains(plan, "table: t1_agg, rollup: mv_t1_v0\n");
```
Comment (LiShuMing) [inline fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvRewriteTest.java:1988]: What was the behavior prior to this pull request?

Could it not rewrite, or did it throw an exception when transformed into a physical plan?

File: fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java [hunk 2]
```diff
@@ -758,11 +758,29 @@ public PlanFragment visitPhysicalOlapScan(OptExpression optExpr, ExecPlan contex
...
+                if (scanBaseTable || !mvColumnMap.containsKey(column.getName())) {
+                    slotDescriptor.setColumn(entry.getValue());
+                    slotDescriptor.setIsNullable(entry.getValue().isAllowNull());
+                } else {
+                    // Replace the column to mv column for special case:
+                    // If the MV is for aggregate table, and the query doesn't contain aggregate function
+                    // The MV would be chosen but the columns would not be rewritten
+                    Column mvColumn = mvColumnMap.get(column.getName());
+                    slotDescriptor.setColumn(mvColumn);
+                    slotDescriptor.setIsNullable(mvColumn.isAllowNull());
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:782]: It seems weird to place this here. Is it possible that the root cause lies within the MaterializedViewRule?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:782]: yes, but it's pretty complicated to fix it in the rule.

The root cause is, previously in statement `CREATE MATERIALIZED VIEW v0 AS SELECT t1_17.c_1_0, t1_17.c_1_1, SUM(t1_17.c_1_2)`, the `SUM(t1_17.c_1_2)` must has an alias, but right now it doesn't, and assigned an automatic name for it.

**Review by satanson — APPROVED** [review]

**Review by LiShuMing — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|14|14|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|14|14|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|14|14|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|14|14|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by mofeiatwork** [comment]
@Mergifyio rebasej

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|14|14|100.00%|[]|

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|14|14|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**